NOnion/Utility/Base32Util.fs

```diff
@@ -0,0 +1,171 @@
+            let a6 = (quintuplet &&& 0x00000003E0L) >>> 5 |> int
+            let a7 = (quintuplet &&& 0x000000001FL) |> int
+
+            match ending with
+            | 1 ->
+                [
+                    alphabet.[a0]
+                    alphabet.[a1]
+                    '='
+                    '='
+                    '='
+                    '='
+                    '='
+                    '='
+                ] // 01======
+            | 2 ->
+                [
+                    alphabet.[a0]
+                    alphabet.[a1]
+                    alphabet.[a2]
+                    alphabet.[a3]
+                    '='
+                    '='
+                    '='
+                    '='
+                ] // 0123====
+            | 3 ->
+                [
+                    alphabet.[a0]
+                    alphabet.[a1]
+                    alphabet.[a2]
+                    alphabet.[a3]
+                    alphabet.[a4]
+                    '='
+                    '='
+                    '='
+                ] // 01234===
+            | 4 ->
+                [
+                    alphabet.[a0]
+                    alphabet.[a1]
+                    alphabet.[a2]
+                    alphabet.[a3]
+                    alphabet.[a4]
+                    alphabet.[a5]
+                    alphabet.[a6]
+                    '='
+                ] // 0123456=
+            | _ ->
+                [
+                    alphabet.[a0]
+                    alphabet.[a1]
+                    alphabet.[a2]
+                    alphabet.[a3]
+                    alphabet.[a4]
+                    alphabet.[a5]
+                    alphabet.[a6]
+                    alphabet.[a7]
+                ] // 01234567
+
+        let rec parse result input =
+            match input with
+            | x0 :: x1 :: x2 :: x3 :: x4 :: tail ->
+                parse (result @ quintupletToList 5 (x0, x1, x2, x3, x4)) tail
+            | x0 :: x1 :: x2 :: [ x3 ] ->
+                result @ quintupletToList 4 (x0, x1, x2, x3, 0uy)
+            | x0 :: x1 :: [ x2 ] ->
+                result @ quintupletToList 3 (x0, x1, x2, 0uy, 0uy)
+            | x0 :: [ x1 ] ->
+                result @ quintupletToList 2 (x0, x1, 0uy, 0uy, 0uy)
+            | [ x0 ] -> result @ quintupletToList 1 (x0, 0uy, 0uy, 0uy, 0uy)
+            | [] -> result
+
+        (text: string)
+        |> System.Text.Encoding.UTF8.GetBytes
+        |> Array.toList
+        |> parse []
+        |> List.toArray
+        |> System.String.Concat
+
+    /// Decodes a Base32 string to a UTF8 string
+    let DecodeBase32 text =
+        if (text: string).Length % 8 <> 0 then
+            [||]
+        else
+            // RFC 4648: The Base 32 alphabet
+            let alphabet =
+                [
+                    for c in "ABCDEFGHIJKLMNOPQRSTUVWXYZ234567=" -> c
+                ]
+                |> List.mapi(fun i a -> a, i)
```

Comment:
@aarani pls longer var names

Reply:
Fixed in [ce823ce0ac6f58340c96927ffea32d8b5a6d628dv](https://github.com/nblockchain/NOnion/pull/30/commits/ce823ce0ac6f58340c96927ffea32d8b5a6d628d).